pkg/config/model/viper.go

```diff
@@ -481,7 +481,11 @@ func (c *safeConfig) UnmarshalExact(rawVal interface{}) error {
 // ReadInConfig wraps Viper for concurrent access
 func (c *safeConfig) ReadInConfig() error {
 	c.Lock()
-	err := c.Viper.ReadInConfig()
+	err := c.configSources[SourceFile].ReadInConfig()
```

Comment:
Can you give more context about what the issue is and how your change fix it?

Reply:
Sure, I updated the PR description!
Apologies for not doing it earlier: I first wanted to open this PR fast to have it backported in 7.50

Reply:
@julesmcrt would be great if we could add tests for these changes. That way in the future if something changes we still keep the wanted behaviour.

Reply:
I added a test that's creating a config file, reading it, then deleting it
Please let me know if creating then deleting files during a test is safe or not
(If not, do you know how to do this test?)

Reply:
I used a tmp file instead, LMK